src/xcode/ENA/ENA/Source/Scenes/Vaccination/VaccinationQRCodeScannerViewModel.swift

```diff
@@ -37,13 +37,22 @@ class VaccinationQRCodeScannerViewModel: NSObject, AVCaptureMetadataOutputObject
 
 		deactivateScanning()
 		guard let code = metadataObjects.first(where: { $0 is MetadataMachineReadableCodeObject }) as? MetadataMachineReadableCodeObject,
-			  let url = code.stringValue,
-			  !url.isEmpty
+			  let fullUrlString = code.stringValue
 		else {
+			Log.error("Vaccination QRCode verification Failed, invalid metadataObject", log: .vaccination)
 			onError?(QRScannerError.codeNotFound)
 			return
 		}
-		self.onSuccess(url)
+		let componentsArray = fullUrlString.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
+		guard fullUrlString.count > 4,
+			componentsArray.first == "HC1",
+			let base45String = componentsArray.last else {
+			Log.error("Vaccination QRCode verification Failed, invalid Prefix", log: .vaccination)
+			onError?(QRScannerError.codeNotFound)
+			return
+		}
```

Comment:
```suggestion
		let componentsArray = fullUrlString.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
		guard fullUrlString.count > 4,
			componentsArray.first == "HC1",
			let base45String = componentsArray.last else {
			Log.error("Vaccination QRCode verification Failed, invalid Prefix", log: .vaccination)
			onError?(QRScannerError.codeNotFound)
			return
		}
```
I think this is a bit more clean